router/io-spec.gobra

```diff
@@ -543,3 +544,103 @@ pure func uInfo(hopfields seq[io.IO_HF], currHFIdx int, consDir bool) set[io.IO_
 						hvfSet(hopfields[currHFIdx]) :
 						hvfSet(hopfields[currHFIdx-1])))
 }
+
+ghost
+decreases
+pure func ifsToIO_ifs(ifs uint16) option[io.IO_ifs]{
+    return ifs == 0 ? none[io.IO_ifs] : some(io.IO_ifs(ifs))
+}
+
+ghost 
+requires acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), _)
+decreases
+pure func validPktMetaHdr(raw []byte) bool {
+	return len(raw) > 4 && 
+		unfolding acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), _) in 
+		let hdr := binary.BigEndian.Uint32(raw[0:4]) in
+		let metaHdr := scion.DecodedFrom(hdr) in
+		let seg1 := int(metaHdr.SegLen[0]) in
+		let seg2 := int(metaHdr.SegLen[1]) in
+		let seg3 := int(metaHdr.SegLen[2]) in
+		let base := scion.Base{metaHdr, 
+			numInfoFields(seg1, seg2, seg3),
+			seg1 + seg2 + seg3} in
+		metaHdr.InBounds() &&
+		0 < metaHdr.SegLen[0] &&
+		base.ValidCurrInfSpec() &&
+		base.ValidCurrHfSpec() &&
+		len(raw) >= pktLen(seg1, seg2, seg3)
+}
+
+// TODO: add body
+ghost 
+requires acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), _)
+ensures val.isIO_val_Unsupported
+ensures val.IO_val_Unsupported_1 == ifsToIO_ifs(ingressID)
+decreases
+pure func absIO_val_Unsupported(raw []byte, ingressID uint16) (ghost val io.IO_val)
+
+ghost
+requires dp.Valid()
+requires acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), _)
+ensures val.isIO_val_Pkt2 || val.isIO_val_Unsupported
+decreases
+pure func absIO_val(dp io.DataPlaneSpec, raw []byte, ingressID uint16) (ghost val io.IO_val) {
+	return (validPktMetaHdr(raw) && absPkt(dp, raw, dp.asid()) != none[io.IO_pkt2]) ? 
+		io.IO_val(io.IO_val_Pkt2{ifsToIO_ifs(ingressID), get(absPkt(dp, raw, dp.asid()))}) : 
+		absIO_val_Unsupported(raw, ingressID)
+}
+
+ghost
+requires acc(&d.localIA, _)
+decreases
+pure func (d *DataPlane) dpSpecWellConfiguredLocalIA(dp io.DataPlaneSpec) bool{
+	return io.IO_as(d.localIA) == dp.Asid()
+}
+
+//TODO: Do we want both implications? 
```

Comment:
Let's drop the comments if you think we do not need it. We can always add it later if necessary